README.md

```diff
@@ -16,9 +16,40 @@ In summary, our software suite addresses the multifaceted challenges associated
+
+Our solution to this trade-off involves splitting the results into three tiers of warnings/errors. The first tier is for best practice violations, which are clear violations of known best practices, such as negative spike times. The second tier is for critical issues, which are potentially critical mistakes, such as zero-rate timestamps. The third tier is for best practice suggestions, which are recommendations for improving the NWB file, such as adding an informative description. This approach allows our inspection process to be as comprehensive as possible while minimizing disruptions from ignoring specific best practice suggestions.
+
+Power users of NWBInspector often seek more direct control of the inspection process and its outputs. To enable more flexibility when using _NWBInspector_, we provide a simple Python API with all the same functionality as the CLI, along with some more advanced features. Just like the CLI, the Python API can be employed to inspect individual NWB files and entire directories of NWB files with the `nwbinspector.inspect_nwbfile()` function and `nwbinspector.inspect_all()` function respectively. Additionally, the Python API allows users to inspect NWB files directly in memory with the `nwbinspector.inspect_nwbfile_object()` function. Since both the `inspect_nwbfile` and the `inspect_all` functions return generators, Python users can employ these functions in more sophisticated ways, such as implementing early stopping if a particular message is raised. These features empower advanced users to fully leverage _NWBInspector_.
+
+Some organizations have their members use NWBInspector, but they may need to adjust the importance level of some checks to meet their requirements. For instance, DANDI requires users to specify the sex and age of their subjects, so those checks should be elevated to importance level "critical" instead of "best practice suggestions." To accommodate this need, NWBInspector provides a configuration option where the importance level of individual checks can be changed with a simple YAML file. Similarly, adding new checks is as simple as defining the check function in some module with a `register_check` decorator, and then specifying the module name with the `--modules` flag in the CLI. For example, in the module defining some new check:
```

Comment:
```suggestion
Some organizations have their members use NWB Inspector, but they may need to adjust the importance level of some checks to meet their requirements. For instance, DANDI requires users to specify the sex and age of their subjects, so those checks should be elevated to importance level "critical" instead of "best practice suggestions." To accommodate this need, NWBInspector provides a configuration option where the importance level of individual checks can be changed with a simple YAML file. Similarly, adding new checks is as simple as defining the check function in some module with a `register_check` decorator, and then specifying the module name with the `--modules` flag in the CLI. For example, in the module defining some new check:
```